### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 21

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#261] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152316646-6047fb9d-9e73-4b03-9dbd-bd6940eb928e.png)

**Comment by LuginiAndrea** [comment]
La prima è vera.
Per il codomonio abbiamo la conferma notando che $$f\left(X\right) \subset Y \cup g\left(Z\right) \subset W \Longrightarrow f\left(X\right) \cup g\left(Z\right) \subset Y \cup W$$ 
Ora verifichiamo il dominio
$$\forall y \in \left(f\left(X\right) \cup g\left(Z\right)\right) \Longrightarrow \exists x \in X \textrm{ t.c } f\left(x\right) = y \vee \exists z \in Z \textrm{ t.c. } g\left(z\right) = y$$
In parole, la controimmagine di $y$ ricade in $X$ o in $Z$, con $y \in f\left(X\right) \cup g\left(Z\right)$. Ergo il dominio della funzione è considerabile $X \cup Z$, quindi abbiamo dimostrato la veridicità della prima ipotesi

La seconda è vera:
Consideriamo $f, g$ nella loro forma insiemistica: Un insieme di tuple $\left(x, y\right)$ per $f$ ed un insieme di tuple $\left(z, w\right)$ per $g$. 
La loro intersezione sarà l'insieme O formato dalle coppie $\left(h, p\right)$ tali che $h = x = z \land p = y = w$.
Ne consegue che $\left(\forall h \in O, h \in X \land h \in Z\right) \land \left(\forall p \in O, p \in Y \land p \in W\right) \Longrightarrow \forall h \in O, h \in X \cap Z \land \forall p \in O, p \in Y \land W$
Dimostrando così l'ipotesi

La terza è falsa in quanto non abbiamo informazioni sulla definizione delle funzioni.
Basti infatti definire 
$$f: \mathbf{Z} \rightarrow \mathbf{Z}, f\left(x\right) = x \times 2$$
$$g: \mathbf{Z} \rightarrow \mathbf{N}, g\left(z\right) = z^2$$
In quanto $\mathbf{Z} \subset \mathbf{Z}$ e $f\left(x\right)$ iniettiva (la dimostrazione è banale) abbiamo rispettato il vincolo dell'ipotesi, ma abbiamo una funzione $g$ non iniettiva (basti controllare con input $-2$ e $2$)
